packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -496,27 +491,20 @@ class WebAssetServer implements AssetReader {
   WebRendererMode webRenderer = WebRendererMode.html;
 
   shelf.Response _serveIndex() {
+
+    final IndexHtml indexHtml = _getIndexHtml();
+
+    indexHtml.applySubstitutions(
+      // Currently, we don't support --base-href for the "run" command.
+      baseHref: '/',
+      serviceWorkerVersion: null,
+    );
+
     final Map<String, String> headers = <String, String>{
       HttpHeaders.contentTypeHeader: 'text/html',
+      HttpHeaders.contentLengthHeader: indexHtml.content.length.toString(),
```

Comment:
I think shelf does this for you. One must be careful with encoding. Did you check before adding this @mdebbar ?

Reply:
Ah! it was already there. Can be removed. I'll do a thing quick..

Reply:
https://github.com/flutter/flutter/pull/119869

Reply:
Good catch @kevmoo, thanks for submitting a patch so quickly!